File: 01. C# Part1/04. ConsoleInputOutput-Homework/06. QuadraticEquation/QuadraticEquation.cs

using System;

    class QuadraticEquation
    {
        static void Main()
        {
            //Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and 
            //solves it (prints its real roots).

            Console.WriteLine("Write your a,b and c for the quadric equation (ax2 + bx + c): ");
            double a = double.Parse(Console.ReadLine());
            double b = double.Parse(Console.ReadLine());
            double c = double.Parse(Console.ReadLine());
            double D = b * b - 4 * a * c;
            double x1 = (-b + Math.Sqrt(D)) / (2 * a);
            double x2 = (-b - Math.Sqrt(D)) / (2 * a);
            Console.WriteLine("X1 = {0}", x1);
            Console.WriteLine("X2 = {0}", x2);
        }
    }

File: 06. JavaScript Basics/03. ConditionalStatements-Homework/03. ConditionalStatements-Homework/06. QuadraticEquation/QuadraticEquation.js
//Write a script that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it 
//(prints its real roots).
//Calculates and prints its real roots.

function quadraticEquation(a, b, c) {
    var d = b * b - 4 * a * c;
       if (a === 0) {
        jsConsole.writeLine('x = ' + (-c / b));
    } else {
        if (d < 0) {
            jsConsole.writeLine('no real roots');
        } else if (d === 0) {
            jsConsole.writeLine('x1 = x2 = ' + ((-b) / (2 * a)));
        } else {
            jsConsole.writeLine('x1 = ' + ((-b - Math.sqrt(d))) / (2 * a));
            jsConsole.writeLine('x2 = ' + ((-b + Math.sqrt(d))) / (2 * a));
        }
    }
}

quadraticEquation(0, 3, 6);
quadraticEquation(2, 2, 2);
quadraticEquation(10, 53, 21);
quadraticEquation(1, 4, 16);